modules/ting_reference/ting_reference.install

```diff
@@ -127,4 +127,60 @@ function ting_reference_update_7001() {
   );
 
   db_create_table('ting_reference', $schema['ting_reference']);
+
+  // Migrate references(materials) field data.
+  _ting_reference_migrate_fields();
+}
+
+/**
+ * Migrate ting reference materials fields.
+ */
+function _ting_reference_migrate_fields() {
+  // Get content types that have materials field.
+  $content_types = field_info_instances('node');
+  $node_types = array();
+  foreach ($content_types as $content_type => $fields) {
+    foreach ($fields as $field) {
+      if (strpos($field['field_name'], 'materials') !== FALSE) {
+        $node_types[$content_type] = $content_type;
+      }
+    }
+  }
+  // Get node ids of the selected types.
+  $nids = array();
+  foreach ($node_types as $node_type) {
+    $result = db_select('node', 'n')
+      ->fields('n', array('nid'))
+      ->condition('type', $node_type)
+      ->execute()
+      ->fetchCol();
+    $nids = array_merge($nids, $result);
+  }
+
+  foreach ($nids as $nid) {
```

Comment:
When you making execute db_api returns a object which already implements iterator interface.